File: fdlp/java/IT_Academy/src/it_academy/Persona.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package it_academy;

/**
 *
 * @author juanj
 */
public class Persona {
    private String nom;
    private int edat;
    private String caracter;
    private Boolean bonaGent;

    public Persona(String nom, int edat, String caracter) {
        this.nom = nom;
        this.edat = edat;
        this.caracter = caracter;
        this.bonaGent=true;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public int getEdat() {
        return edat;
    }

    public void setEdat(int edat) {
        this.edat = edat;
    }

    public String getCaracter() {
        return caracter;
    }

    public void setCaracter(String caracter) {
        this.caracter = caracter;
    }

    public Boolean getBonaGent() {
        return bonaGent;
    }

    public void setBonaGent(Boolean bonaGent) {
        this.bonaGent = bonaGent;
    }
    
    
    public int creixer(int anys){
        return this.edat+anys;
    }
    
    
    
    
    
}

File: fdlp/Recursos/POO/segon_exemple/persona.js
class Persona{
    
    constructor(nom, edat, caracter){
        this._nom = nom;
        this._edat = edat;
        this._caracter = caracter 
        this._bonaGent = true;
    }
    
    
    //GETTERS i SETTERS
    
    
    getNom(){
        return this._nom;
    }
    
    setNom(x){
        this._nom = x;
    }

    getEdat(){
        return this._edat;
    }

    setEdat(edat){
        this._edat = edat;
    }
    
    //get getEdat(){                     //el nom del get, definirà el objecte.nom
    //    return this._edat;
    //}




    //set setEdat(edat){
    //    this._edat = edat;
    //}
    




    set setEdat(e){                //el nom del get, definirà el objecte.nom
        this._edat = e;
    }
    
    get caracter(){
        return this._caracter;
    }

    //set setCaracter(formaDeActuar){
    //    this._caracter = formaDeActuar;
    //}

    setCaracter(formaDeActuar){
    this._caracter = formaDeActuar;
    }
   
    //METODES DE LA CLASSE
    parlar(paraula){
        console.log("Estic dient: " +paraula);
    }
    
    caminar(){
        return "Estic caminant";
    }

    creixer(numAnys){
        console.log("Abans tenia " + this._edat+ " anys");
        this._edat +=  numAnys;
        console.log("Ara tinc " + this._edat+ " anys");
    }

}